artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPConnectionContext.java

```diff
@@ -87,9 +88,7 @@
 
    private final boolean useCoreSubscriptionNaming;
 
-   private boolean isSchedulingCancelled;
-   private ScheduledFuture scheduledFuture;
-   private final Object schedulingLock = new Object();
+   private AtomicReference scheduledFutureRef;
```

Comment:
make it final and it would be better to use the not raw version by using <>. You can always create a singleton instance to be used to do your logic

Reply:
I made it final. How could I use a singleton instance in this case?